gen/generator.go

```diff
@@ -139,19 +140,36 @@ type generator struct {
 	// TODO use something to group related decls together
 }
 
+type GeneratorOptions struct {
+	TImporter   ThriftPackageImporter
+	ImportPath  string
+	PackageName string
+
+	NoZap bool
+}
+
 // NewGenerator sets up a new generator for Go code.
-func NewGenerator(timport ThriftPackageImporter, importPath string, packageName string) Generator {
+func NewGenerator(o *GeneratorOptions) Generator {
 	// TODO(abg): Determine package name from `namespace go` directive.
 	namespace := NewNamespace()
 	return &generator{
-		PackageName:    packageName,
-		ImportPath:     importPath,
+		PackageName:    o.PackageName,
+		ImportPath:     o.ImportPath,
 		Namespace:      namespace,
 		importer:       newImporter(namespace.Child()),
 		mangler:        newMangler(),
-		thriftImporter: timport,
+		thriftImporter: o.TImporter,
 		fset:           token.NewFileSet(),
+		noZap:          o.NoZap,
+	}
+}
+
+// checkNoZap returns whether the NoZap flag is passed.
+func checkNoZap(g Generator) bool {
+	if castedGenerator, ok := g.(*generator); ok {
```

Comment:
nit: shorter variable names are okay when there are only 2-3 variables in scope. `gen` would be fine here.